### DISCUSSION TITLE
Problem with updated gcc11 in bigsur/arm64

### DISCUSSION THREAD

**Discussion by mottelet** [Homebrew/discussions#1490] [Casks]
Hello,

My gcc has been automatically updated to gcc11 (was gcc10 before), and now during compilation of my software (Scilab) I get

~~~~
ld: file not found: @rpath/libgcc_s.1.1.dylib for architecture arm64
~~~~

Compilation was ok before (with gcc10).

**Comment by carlocab** [comment]
`libgcc_s.1.1.dylib` is now located at `$(brew --prefix gcc)/lib/gcc/11` so you'll need to adjusted your linker search path.

This (or the equivalent) should previously have been found at `$(brew --prefix gcc)/lib/gcc/10`.

**Comment by mottelet** [comment]
My configure script detected the right version but maybe the ld cache has to be cleared (what is the command ?)

S.

**Reply by carlocab** [reply]
What's the output of
```
file $(brew --prefix gcc)/lib/gcc/11/libgcc_s.1.1.dylib
```
?

**Reply by mottelet** [reply]
~~~~
/opt/homebrew/opt/gcc/lib/gcc/11/libgcc_s.1.1.dylib: Mach-O universal binary with 1 architecture: [arm64:Mach-O 64-bit dynamically linked shared library arm64]
/opt/homebrew/opt/gcc/lib/gcc/11/libgcc_s.1.1.dylib (for architecture arm64):	Mach-O 64-bit dynamically linked shared library arm64
~~~~

**Comment by mottelet** [comment]
I am really sorry but I still get this error (here with more context)
~~~~
ld: warning: dylib (/opt/homebrew/opt/gcc/lib/gcc/11/libgfortran.5.dylib) was built for newer macOS version (11.3) than being linked (11.0)
ld: file not found: @rpath/libgcc_s.1.1.dylib for architecture arm64
~~~~
It seems that is when linking the gfortran lib that libgcc_s.1.1.dylib is not found. And I repeat, everything was fine before the update from gcc-10 to gcc-11.

S.

**Reply by carlocab** [reply]
Can you try the following, and tell me if it helps:
```
cd $(brew --prefix gcc)/lib/gcc/11
lipo libgcc_s.1.1.dylib -thin arm64 -output libgcc_s.1.1.dylib
```

**Comment by mottelet** [comment]
Hi,

fatal error: 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: 
input file (libgcc_s.1.1.dylib) must be a fat file when the -thin option 
is specified

I solved the compilation by adding

LDFLAGS=-L/opt/homebrew/opt/gcc/lib/gcc/11

to the configure script. It was not necessary before.

S.

---
Stephane Mottelet

Le 2021-05-19 10:59, Carlo Cabrera a écrit :

> Can you try the following, and tell me if it helps:
> 
> cd $(brew --prefix gcc)/lib/gcc/11
> lipo libgcc_s.1.1.dylib -thin arm64 -output libgcc_s.1.1.dylib
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].


Links:
------
[1] 
https://antispam.utc.fr/proxy/2/c3RlcGhhbmUubW90dGVsZXRAdXRjLmZy/github.com/Homebrew/discussions/discussions/1490#discussioncomment-757193
[2]

**Reply by carlocab** [reply]
What's the output of
```
file $(brew --prefix gcc)/lib/gcc/11/libgcc_s.1.1.dylib
```
now?

Your `configure` script probably knew how to find `gcc-10`, but doesn't know how to find `gcc-11`, which is why you need to set `LDFLAGS` now but not previously.

**Comment by mottelet** [comment]
/opt/homebrew/opt/gcc/lib/gcc/11/libgcc_s.1.1.dylib: Mach-O 64-bit 
dynamically linked shared library arm64

I guess there is a library cache to be flushed somewhere since the 
problem is not at configure time but only at link time.

S.

---
Stephane Mottelet

Le 2021-05-19 12:41, Carlo Cabrera a écrit :

> What's the output of
> 
> file $(brew --prefix gcc)/lib/gcc/11/libgcc_s.1.1.dylib
> 
> now?
> 
> Your configure script probably knew how to find gcc-10, but doesn't 
> know how to find gcc-11, which is why you need to set LDFLAGS now but 
> not previously.
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].


Links:
------
[1] 
https://antispam.utc.fr/proxy/2/c3RlcGhhbmUubW90dGVsZXRAdXRjLmZy/github.com/Homebrew/discussions/discussions/1490#discussioncomment-757662
[2]

**Reply by carlocab** [reply]
Sounds unlikely. It looks like your configure time check didn't work, as it was looking for `@rpath/libgcc_s.1.1.dylib`. That's not even the install name of `libgcc_s.1.1.dylib` (even back when it was GCC 10).

**Comment by mottelet** [comment]
Hello,

You are not right, this is the actual install name of libgcc_s.1.1.dylib 
in libgfortran.5.dylib that was installed with homebrew gcc11 package:

***@***.*** 11 % otool -L 
/opt/homebrew/opt/gcc/lib/gcc/11/libgfortran.5.dylib

/opt/homebrew/opt/gcc/lib/gcc/11/libgfortran.5.dylib:

  /opt/homebrew/opt/gcc/lib/gcc/11/libgfortran.5.dylib (compatibility 
version 6.0.0, current version 6.0.0)

  @rpath/libgcc_s.1.1.dylib (compatibility version 1.0.0, current version 
1.1.0)

  /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current 
version 1292.100.5)

S.

---
Stephane Mottelet

Le 2021-05-20 02:36, Carlo Cabrera a écrit :

> Sounds unlikely. It looks like your configure time check didn't work, 
> as it was looking for @rpath/libgcc_s.1.1.dylib. That's not even the 
> install name of libgcc_s.1.1.dylib (even back when it was GCC 10).
> 
> --
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].


Links:
------
[1] 
https://antispam.utc.fr/proxy/2/c3RlcGhhbmUubW90dGVsZXRAdXRjLmZy/github.com/Homebrew/discussions/discussions/1490#discussioncomment-761045
[2]

**Reply by carlocab** [reply]
That seems off, since
```
otool -D libgfortran.5.dylib:
/opt/homebrew/opt/gcc/lib/gcc/11/libgfortran.5.dylib
```
In any case:
```
❯ otool -l libgfortran.5.dylib | rg -A2 LC_RPATH
          cmd LC_RPATH
      cmdsize 32
         path @loader_path (offset 12)
```

**Reply by carlocab** [reply]
Sorry, that `otool -D` should've been
```
❯ otool -D libgcc_s.1.1.dylib
libgcc_s.1.1.dylib (architecture arm64):
/opt/homebrew/opt/gcc/lib/gcc/11/libgcc_s.1.1.dylib
```
